src/storage/datacore/restapi/mode/statusmonitor.pm

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright 2024 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+package storage::datacore::restapi::mode::statusmonitor;
+use strict;
+use warnings;
+
+use base qw(centreon::plugins::templates::counter);
+use JSON::XS;
+use Digest::MD5 qw(md5_hex);
+use centreon::plugins::templates::catalog_functions qw(catalog_status_threshold_ng);
+use centreon::plugins::misc qw(empty);
+
+my %monitor_state = (1 => "Undefined", 2 => "Healthy", 4 => "Attention", 8 => "Warning", 16 => "Critical");
+
+sub new {
+    my ($class, %options) = @_;
+
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+    $options{options}->add_options(arguments => {
+        'filter-caption:s' => { name => 'filter_caption' } });
+    $self->{cache} = centreon::plugins::statefile->new(%options);
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+    $self->{cache}->check_options(option_results => $self->{option_results});
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 1, message_multiple => 'All memory usages are ok', skipped_code => { -10 => 1 } }
```

Comment:
I think probably it's most a 0 type than a 1 (like [here](https://github.com/centreon/centreon-plugins/blob/c245c4104160636d900256ba0abd1bc836437467/src/hardware/sensors/rittal/cmc3/snmp/mode/unitstatus.pm#L74))

Reply:
To be discussed together, but there can be multiples "monitors" to check, that why I made a for loop in manage_selection

Reply:
I haven't see the instance to handle my bad